data/src/client.rs

```diff
@@ -57,6 +62,7 @@ pub enum Event {
     Single(message::Encoded, Nick),
     WithTarget(message::Encoded, Nick, message::Target),
     Brodcast(Brodcast),
+    Notification(Notification),
```

Comment:
We need to add `message::Encoded` here so we still record the message in history once returned from here